Part 2: Does not 'make' with Ubuntu 14.04.5 LTS 32bits

Dear @vsonnier

We still have errors:

```
Dev/CubicSDR/build$ make
Scanning dependencies of target CubicSDR
[  1%] Building CXX object CMakeFiles/CubicSDR.dir/external/rtaudio/RtAudio.cpp.o
[  2%] Building CXX object CMakeFiles/CubicSDR.dir/src/CubicSDR.cpp.o
In file included from ~/Dev/CubicSDR/external/cubicvr2/math/aabb.h:13:0,
                 from ~/Dev/CubicSDR/external/cubicvr2/math/cubic_math.h:14,
                 from ~/Dev/CubicSDR/src/ui/GLPanel.h:10,
                 from ~/Dev/CubicSDR/src/panel/ScopePanel.h:6,
                 from ~/Dev/CubicSDR/src/process/ScopeVisualProcessor.h:8,
                 from ~/Dev/CubicSDR/src/visual/ScopeCanvas.h:13,
                 from ~/Dev/CubicSDR/src/AppFrame.h:15,
                 from ~/Dev/CubicSDR/src/CubicSDR.h:22,
                 from ~/Dev/CubicSDR/src/CubicSDR.cpp:17:
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h: In static member function ‘static CubicVR::__float CubicVR::vec3::length(CubicVR::vec3, CubicVR::vec3)’:
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:47:22: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             a = ptb[0]-pta[0];
                      ^
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:33:18: note: candidate 1: CubicVR::__float& CubicVR::vec3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:47:22: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             a = ptb[0]-pta[0];
                      ^
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:47:29: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             a = ptb[0]-pta[0];
                             ^
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:33:18: note: candidate 1: CubicVR::__float& CubicVR::vec3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:47:29: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             a = ptb[0]-pta[0];
                             ^
~/Dev/CubicSDR/external/cubicvr2/math/vec3.h:48:22: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             b = ptb[1]-pta[1];
                      ^
(... many more errors...)

In file included from ~/Dev/CubicSDR/external/cubicvr2/math/cubic_math.h:16:0,
                 from ~/Dev/CubicSDR/src/ui/GLPanel.h:10,
                 from ~/Dev/CubicSDR/src/panel/ScopePanel.h:6,
                 from ~/Dev/CubicSDR/src/process/ScopeVisualProcessor.h:8,
                 from ~/Dev/CubicSDR/src/visual/ScopeCanvas.h:13,
                 from ~/Dev/CubicSDR/src/AppFrame.h:15,
                 from ~/Dev/CubicSDR/src/CubicSDR.h:22,
                 from ~/Dev/CubicSDR/src/CubicSDR.cpp:17:
~/Dev/CubicSDR/external/cubicvr2/math/mat4.h:123:53: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             return mat4(m[0], m[4], m[8], m[12], m[1], m[5], m[9], m[13], m[2], m[6], m[10], m[14], m[3], m[7], m[11], m[15]);
                                                     ^
^Cmake[2]: *** [CMakeFiles/CubicSDR.dir/src/CubicSDR.cpp.o] Interrupt
make[1]: *** [CMakeFiles/CubicSDR.dir/all] Interrupt
make: *** [all] Interrupt
```

make CubicSDR-0.2.2 (just a few parts of the log, because it has more than 100MB)

Notice that this time 'make' does not abort and finishes, but with many erros

```
~Dev/CubicSDR-0.2.2/build$ cmake ../ -DCMAKE_BUILD_TYPE=Release -DwxWidgets_CONFIG_EXECUTABLE=~/Dev/wxWidgets-staticlib/bin/wx-config
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
32 bit compiler detected
-- Looking for XOpenDisplay in /usr/lib/i386-linux-gnu/libX11.so;/usr/lib/i386-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/i386-linux-gnu/libX11.so;/usr/lib/i386-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/i386-linux-gnu/libX11.so
-- Found OpenGL: /usr/lib/i386-linux-gnu/libGL.so  
-- Found LIQUID: /usr/local/lib/libliquid.so  
-- Found wxWidgets: TRUE  
-- Copying ~/Dev/CubicSDR-0.2.2/font/*.fnt to directory ~/Dev/CubicSDR-0.2.2/build/x86/fonts
-- Configuring file vera_sans_mono36.fnt
-- Configuring file vera_sans_mono24.fnt
-- Configuring file vera_sans_mono18.fnt
-- Configuring file vera_sans_mono32.fnt
-- Configuring file vera_sans_mono12.fnt
-- Configuring file vera_sans_mono16.fnt
-- Configuring file vera_sans_mono48.fnt
-- Configuring file vera_sans_mono64.fnt
-- Configuring file vera_sans_mono72.fnt
-- Configuring file vera_sans_mono96.fnt
-- Configuring file vera_sans_mono27.fnt
-- Copying ~/Dev/CubicSDR-0.2.2/font/*.png to directory ~/Dev/CubicSDR-0.2.2/build/x86/fonts
-- Configuring file vera_sans_mono36_0.png
-- Configuring file vera_sans_mono64_0.png
-- Configuring file vera_sans_mono32_0.png
-- Configuring file vera_sans_mono72_0.png
-- Configuring file vera_sans_mono48_0.png
-- Configuring file vera_sans_mono24_0.png
-- Configuring file vera_sans_mono12_0.png
-- Configuring file vera_sans_mono18_0.png
-- Configuring file vera_sans_mono16_0.png
-- Configuring file vera_sans_mono96_0.png
-- Configuring file vera_sans_mono27_0.png
-- Copying ~/Dev/CubicSDR-0.2.2/icon/CubicSDR.ico to directory ~/Dev/CubicSDR-0.2.2/build/x86
-- Configuring file CubicSDR.ico
-- Configuring done
-- Generating done
-- Build files have been written to: ~/Dev/CubicSDR-0.2.2/build

cmake finished OK

/Dev/CubicSDR-0.2.2/build$ make
Scanning dependencies of target CubicSDR
[  1%] Building CXX object CMakeFiles/CubicSDR.dir/external/rtaudio/RtAudio.cpp.o
[  2%] Building CXX object CMakeFiles/CubicSDR.dir/src/CubicSDR.cpp.o
In file included from ~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/aabb.h:13:0,
                 from ~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/cubic_math.h:14,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/GLPanel.h:10,
                 from ~/Dev/CubicSDR-0.2.2/src/panel/ScopePanel.h:6,
                 from ~/Dev/CubicSDR-0.2.2/src/process/ScopeVisualProcessor.h:8,
                 from ~/Dev/CubicSDR-0.2.2/src/visual/ScopeCanvas.h:13,
                 from ~/Dev/CubicSDR-0.2.2/src/AppFrame.h:15,
                 from ~/Dev/CubicSDR-0.2.2/src/CubicSDR.h:22,
                 from ~/Dev/CubicSDR-0.2.2/src/CubicSDR.cpp:17:
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/vec3.h: In static member function ‘static CubicVR::__float CubicVR::vec3::length(CubicVR::vec3, CubicVR::vec3)’:
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/vec3.h:47:22: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             a = ptb[0]-pta[0];
                      ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/vec3.h:33:18: note: candidate 1: CubicVR::__float& CubicVR::vec3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/vec3.h:47:22: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             a = ptb[0]-pta[0];
                      ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/vec3.h:47:29: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             a = ptb[0]-pta[0];
                             ^
***** (many similar errors) *****

~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h: In static member function ‘static CubicVR::mat3 CubicVR::mat3::transpose(CubicVR::mat3)’:
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:58:35: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             __float a01 = mat_in[1], a02 = mat_in[2], a12 = mat_in[5];
                                   ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:58:35: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             __float a01 = mat_in[1], a02 = mat_in[2], a12 = mat_in[5];
                                   ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:58:52: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             __float a01 = mat_in[1], a02 = mat_in[2], a12 = mat_in[5];
                                                    ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:58:52: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             __float a01 = mat_in[1], a02 = mat_in[2], a12 = mat_in[5];
                                                    ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:58:69: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             __float a01 = mat_in[1], a02 = mat_in[2], a12 = mat_in[5];
                                                                     ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:58:69: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             __float a01 = mat_in[1], a02 = mat_in[2], a12 = mat_in[5];
                                                                     ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:62:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[1] = mat_in[3];
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:62:18: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[1] = mat_in[3];
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:62:30: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[1] = mat_in[3];
                              ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:62:30: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[1] = mat_in[3];
                              ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:63:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[2] = mat_in[6];
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:63:18: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[2] = mat_in[6];
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:63:30: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[2] = mat_in[6];
                              ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:63:30: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[2] = mat_in[6];
                              ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:64:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[3] = a01;
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:64:18: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[3] = a01;
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:65:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[5] = mat_in[7];
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:65:18: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[5] = mat_in[7];
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:65:30: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[5] = mat_in[7];
                              ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:65:30: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[5] = mat_in[7];
                              ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:66:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[6] = a02;
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:66:18: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[6] = a02;
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:67:18: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mat[7] = a12;
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:27:18: note: candidate 1: CubicVR::__float& CubicVR::mat3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:67:18: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             mat[7] = a12;
                  ^
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h: In static member function ‘static CubicVR::vec3 CubicVR::mat3::multiply(CubicVR::mat3, CubicVR::vec3)’:
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/mat3.h:75:19: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             mOut[0] = m2[0] * m1[0] + m2[3] * m1[1] + m2[6] * m1[2] ;
                   ^

***** (many similar errors) *****


~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/frustum.h:135:66: warning: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [enabled by default]
             points[6] = vec3(bbox.max[0], bbox.max[1], bbox.min[2]);
                                                                  ^
In file included from ~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/aabb.h:13:0,
                 from ~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/cubic_math.h:14,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/GLPanel.h:10,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/UITestContext.h:7,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/UITestContext.cpp:4:
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/vec3.h:33:18: note: candidate 1: CubicVR::__float& CubicVR::vec3::operator[](unsigned int)
         __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
                  ^
In file included from ~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/cubic_math.h:25:0,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/GLPanel.h:10,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/UITestContext.h:7,
                 from ~/Dev/CubicSDR-0.2.2/src/ui/UITestContext.cpp:4:
~/Dev/CubicSDR-0.2.2/external/cubicvr2/math/frustum.h:135:66: note: candidate 2: operator[](CubicVR::__float* {aka float*}, int) <built-in>
             points[6] = vec3(bbox.max[0], bbox.max[1], bbox.min[2]);
                                                                  ^
[100%] Linking CXX executable x86/CubicSDR
[100%] Built target CubicSDR
```

the binary resulting from make does run and open GUI, but since my RSP2 did not arrive yet, I cannot test it to see if it can detected it.

I do have a NooElec NESDR Nano 2+ Tiny Black RTL-SDR USB, but since I followed the "steps" for sdrplay I understand that it does not detect it.

Conclusion there is a syntax problem on the sources in respect with the 32bit version I have installed here, which are:

-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4

I also understand that @starvald and @frspin are running on 64bits version.

So could it be a problem of 64bits versus 32bits ?

@vsonnier  just did the make.  Good job! Much better this time, but still some glitches.  Please see below the output for you analysis. 

Must say, you are excellent coder (and quick - based on my undersanding of C++ and the number of coded lines of this project).

```
~/Downloads/SDR/CubicSDR-VR2/build$ cmake ../ -DCMAKE_BUILD_TYPE=Release -DwxWidgets_CONFIG_EXECUTABLE=~/Downloads/SDR/Dev/wxWidgets-staticlib/bin/wx-config
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
32 bit compiler detected
-- Looking for XOpenDisplay in /usr/lib/i386-linux-gnu/libX11.so;/usr/lib/i386-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/i386-linux-gnu/libX11.so;/usr/lib/i386-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/i386-linux-gnu/libX11.so
-- Found OpenGL: /usr/lib/i386-linux-gnu/libGL.so  
-- Found LIQUID: /usr/local/lib/libliquid.so  
-- Found wxWidgets: TRUE  
-- Copying ~/Downloads/SDR/CubicSDR-VR2/font/*.fnt to directory ~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts
-- Configuring file vera_sans_mono36.fnt
-- Configuring file vera_sans_mono24.fnt
-- Configuring file vera_sans_mono18.fnt
-- Configuring file vera_sans_mono32.fnt
-- Configuring file vera_sans_mono12.fnt
-- Configuring file vera_sans_mono16.fnt
-- Configuring file vera_sans_mono48.fnt
-- Configuring file vera_sans_mono64.fnt
-- Configuring file vera_sans_mono72.fnt
-- Configuring file vera_sans_mono96.fnt
-- Configuring file vera_sans_mono27.fnt
-- Copying ~/Downloads/SDR/CubicSDR-VR2/font/*.png to directory ~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts
-- Configuring file vera_sans_mono36_0.png
-- Configuring file vera_sans_mono64_0.png
-- Configuring file vera_sans_mono32_0.png
-- Configuring file vera_sans_mono72_0.png
-- Configuring file vera_sans_mono48_0.png
-- Configuring file vera_sans_mono24_0.png
-- Configuring file vera_sans_mono12_0.png
-- Configuring file vera_sans_mono18_0.png
-- Configuring file vera_sans_mono16_0.png
-- Configuring file vera_sans_mono96_0.png
-- Configuring file vera_sans_mono27_0.png
-- Copying ~/Downloads/SDR/CubicSDR-VR2/icon/CubicSDR.ico to directory ~/Downloads/SDR/CubicSDR-VR2/build/x86
-- Configuring file CubicSDR.ico
-- Configuring done
-- Generating done
-- Build files have been written to: ~/Downloads/SDR/CubicSDR-VR2/build
john@john-965P-DS3:~/Downloads/SDR/CubicSDR-VR2/build$ make
Scanning dependencies of target CubicSDR
[  1%] Building CXX object CMakeFiles/CubicSDR.dir/external/rtaudio/RtAudio.cpp.o
[  2%] Building CXX object CMakeFiles/CubicSDR.dir/src/CubicSDR.cpp.o
[  3%] Building CXX object CMakeFiles/CubicSDR.dir/src/AppFrame.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/AppFrame.cpp:1330:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 bool AppFrame::actionOnMenuRig(wxCommandEvent& event) {
      ^
~/Downloads/SDR/CubicSDR-VR2/src/AppFrame.cpp:2010:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void AppFrame::OnAboutDialogClose(wxCommandEvent& event) {
      ^
~/Downloads/SDR/CubicSDR-VR2/src/AppFrame.cpp:2579:6: warning: unused parameter ‘center_freq’ [-Wunused-parameter]
 void AppFrame::setViewState(long long center_freq) {
      ^
~/Downloads/SDR/CubicSDR-VR2/src/AppFrame.cpp: In function ‘std::vector<std::basic_string<char> > str_explode(const string&, const string&)’:
~/Downloads/SDR/CubicSDR-VR2/src/AppFrame.cpp:2611:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (j == std::string::npos && i < str_len)  j = str_len;
                               ^
~/Downloads/SDR/CubicSDR-VR2/src/AppFrame.cpp:2613:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (j == std::string::npos) break;
                               ^
[  5%] Building CXX object CMakeFiles/CubicSDR.dir/src/AppConfig.cpp.o
[  6%] Building CXX object CMakeFiles/CubicSDR.dir/src/FrequencyDialog.cpp.o
[  7%] Building CXX object CMakeFiles/CubicSDR.dir/src/DemodLabelDialog.cpp.o
[  8%] Building CXX object CMakeFiles/CubicSDR.dir/src/IOThread.cpp.o
[ 10%] Building CXX object CMakeFiles/CubicSDR.dir/src/ModemProperties.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/ModemProperties.cpp: In member function ‘void ModemProperties::initDefaultProperties()’:
~/Downloads/SDR/CubicSDR-VR2/src/ModemProperties.cpp:76:13: warning: variable ‘defaultDevice’ set but not used [-Wunused-but-set-variable]
         int defaultDevice = 0;
             ^
~/Downloads/SDR/CubicSDR-VR2/src/ModemProperties.cpp: At global scope:
~/Downloads/SDR/CubicSDR-VR2/src/ModemProperties.cpp:264:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ModemProperties::OnCollapse(wxPropertyGridEvent &event) {
      ^
~/Downloads/SDR/CubicSDR-VR2/src/ModemProperties.cpp:268:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ModemProperties::OnExpand(wxPropertyGridEvent &event) {
      ^
[ 11%] Building CXX object CMakeFiles/CubicSDR.dir/src/BookmarkMgr.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/BookmarkMgr.cpp:502:18: warning: unused parameter ‘name_in’ [-Wunused-parameter]
 BookmarkEntryPtr BookmarkMgr::nodeToBookmark(const char *name_in, DataNode *node) {
                  ^
[ 12%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SDRDeviceInfo.cpp.o
[ 13%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SDRPostThread.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/sdr/SDRPostThread.cpp: In member function ‘void SDRPostThread::updateActiveDemodulators()’:
~/Downloads/SDR/CubicSDR-VR2/src/sdr/SDRPostThread.cpp:98:38: warning: unused variable ‘demodQueue’ [-Wunused-variable]
         DemodulatorThreadInputQueue *demodQueue = demod->getIQInputDataPipe();
                                      ^
[ 15%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SDREnumerator.cpp.o
[ 16%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorPreThread.cpp.o
[ 17%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorThread.cpp.o
[ 18%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorWorkerThread.cpp.o
[ 20%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorInstance.cpp.o
[ 21%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorMgr.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/demod/DemodulatorMgr.cpp: In member function ‘DemodulatorInstance* DemodulatorMgr::loadInstance(DataNode*)’:
~/Downloads/SDR/CubicSDR-VR2/src/demod/DemodulatorMgr.cpp:529:10: warning: variable ‘found_device’ set but not used [-Wunused-but-set-variable]
     bool found_device = false;
          ^
[ 22%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/Modem.cpp.o
[ 23%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/ModemAnalog.cpp.o
[ 25%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/ModemDigital.cpp.o
[ 26%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemAM.cpp.o
[ 27%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemDSB.cpp.o
[ 28%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemFM.cpp.o
[ 30%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemNBFM.cpp.o
[ 31%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemFMStereo.cpp.o
[ 32%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemIQ.cpp.o
[ 33%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemLSB.cpp.o
[ 35%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemUSB.cpp.o
[ 36%] Building CXX object CMakeFiles/CubicSDR.dir/src/audio/AudioThread.cpp.o
[ 37%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/Gradient.cpp.o
[ 38%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/Timer.cpp.o
[ 40%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/MouseTracker.cpp.o
[ 41%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/GLExt.cpp.o
[ 42%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/GLFont.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/util/GLFont.cpp:15:21: warning: ‘std::wstring getExePath()’ defined but not used [-Wunused-function]
 static std::wstring getExePath(void)
                     ^
[ 43%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/DataTree.cpp.o
[ 45%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/ScopePanel.cpp.o
[ 46%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/SpectrumPanel.cpp.o
[ 47%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/WaterfallPanel.cpp.o
[ 48%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/MeterPanel.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/panel/MeterPanel.cpp:129:7: warning: unused parameter ‘panel’ [-Wunused-parameter]
 float MeterPanel::getMeterHitValue(CubicVR::vec2 mousePoint, GLPanel &panel) {
       ^
[ 50%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ColorTheme.cpp.o
[ 51%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/PrimaryGLContext.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/visual/PrimaryGLContext.cpp: In member function ‘void PrimaryGLContext::DrawDemod(DemodulatorInstance*, RGBA4f, long long int, long long int)’:
~/Downloads/SDR/CubicSDR-VR2/src/visual/PrimaryGLContext.cpp:359:19: warning: variable ‘demodAlign’ set but not used [-Wunused-but-set-variable]
     GLFont::Align demodAlign = GLFont::GLFONT_ALIGN_CENTER;
                   ^
[ 52%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/InteractiveCanvas.cpp.o
[ 53%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/MeterCanvas.cpp.o
[ 55%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/MeterContext.cpp.o
[ 56%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/TuningCanvas.cpp.o
[ 57%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/TuningContext.cpp.o
[ 58%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ModeSelectorCanvas.cpp.o
[ 60%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ModeSelectorContext.cpp.o
[ 61%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ScopeCanvas.cpp.o
[ 62%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ScopeContext.cpp.o
[ 63%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/SpectrumCanvas.cpp.o
[ 65%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/WaterfallCanvas.cpp.o
[ 66%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/GainCanvas.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/visual/GainCanvas.cpp: In member function ‘void GainCanvas::updateGainUI()’:
~/Downloads/SDR/CubicSDR-VR2/src/visual/GainCanvas.cpp:206:16: warning: deleting object of polymorphic class type ‘MeterPanel’ which has non-virtual destructor might cause undefined behaviour [-Wdelete-non-virtual-dtor]
         delete mDel;
                ^
[ 67%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ImagePanel.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/visual/ImagePanel.cpp:13:6: warning: unused parameter ‘evt’ [-Wunused-parameter]
 void ImagePanel::paintEvent(wxPaintEvent & evt) {
      ^
[ 68%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/VisualProcessor.cpp.o
[ 70%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/ScopeVisualProcessor.cpp.o
[ 71%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/SpectrumVisualProcessor.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/process/SpectrumVisualProcessor.cpp: In member function ‘virtual void SpectrumVisualProcessor::process()’:
~/Downloads/SDR/CubicSDR-VR2/src/process/SpectrumVisualProcessor.cpp:418:17: warning: ‘bwDiff’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 if (bwDiff < 0) {
                 ^
[ 72%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/FFTVisualDataThread.cpp.o
[ 73%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/FFTDataDistributor.cpp.o
[ 75%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/SpectrumVisualDataThread.cpp.o
[ 76%] Building CXX object CMakeFiles/CubicSDR.dir/src/ui/GLPanel.cpp.o
[ 77%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDevices.cpp.o
[ 78%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDevicesForm.cpp.o
[ 80%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDeviceAdd.cpp.o
[ 81%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDeviceAddForm.cpp.o
[ 82%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Bookmark/BookmarkPanel.cpp.o
[ 83%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Bookmark/BookmarkView.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/forms/Bookmark/BookmarkView.cpp:676:11: warning: unused parameter ‘parent’ [-Wunused-parameter]
 wxButton *BookmarkView::makeButton(wxWindow *parent, std::string labelVal, wxObjectEventFunction handler) {
           ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Bookmark/BookmarkView.cpp:1475:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void BookmarkView::onSearchText( wxCommandEvent& event ) {
      ^
[ 85%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/ActionDialogBase.cpp.o
[ 86%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/ActionDialog.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/ActionDialog.cpp:39:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ActionDialog::onClickCancel( wxCommandEvent& event ) {
      ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/ActionDialog.cpp:48:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ActionDialog::onClickOK( wxCommandEvent& event ) {
      ^
[ 87%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/AboutDialogBase.cpp.o
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp: In constructor ‘AboutDialogBase::AboutDialogBase(wxWindow*, wxWindowID, const wxString&, const wxPoint&, const wxSize&, long int)’:
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:25:67: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_appName->SetFont( wxFont( 20, 70, 90, 90, false, wxEmptyString ) );
                                                                   ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:48:68: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dbHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                    ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:54:97: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dbGHHeader->SetFont( wxFont( wxNORMAL_FONT->GetPointSize(), 70, 90, 90, false, wxEmptyString ) );
                                                                                                 ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:60:96: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dbTwitter->SetFont( wxFont( wxNORMAL_FONT->GetPointSize(), 70, 90, 90, false, wxEmptyString ) );
                                                                                                ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:101:79: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_cContributorsHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                               ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:107:94: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_cGitHub->SetFont( wxFont( wxNORMAL_FONT->GetPointSize(), 70, 90, 90, false, wxEmptyString ) );
                                                                                              ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:193:67: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                   ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:386:68: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                    ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:395:82: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stSoapyDevAssistHeader->SetFont( wxFont( 10, 70, 90, 92, false, wxEmptyString ) );
                                                                                  ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:408:77: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stLiquidDSPHeader->SetFont( wxFont( 10, 70, 90, 92, false, wxEmptyString ) );
                                                                             ^
~/Downloads/SDR/CubicSDR-VR2/src/forms/Dialog/AboutDialogBase.cpp:421:83: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stIdeasDirectionsHeader->SetFont( wxFont( 10, 70, 90, 92, false, wxEmptyString ) );
                                                                                   ^
[ 88%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/AboutDialog.cpp.o
[ 90%] Building CXX object CMakeFiles/CubicSDR.dir/external/lodepng/lodepng.cpp.o
[ 91%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinyxml.cpp.o
[ 92%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinystr.cpp.o
[ 93%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinyxmlparser.cpp.o
[ 95%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinyxmlerror.cpp.o
[ 96%] Building CXX object CMakeFiles/CubicSDR.dir/external/cubicvr2/math/cubic_math.cpp.o
[ 97%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SoapySDRThread.cpp.o
[ 98%] Building CXX object CMakeFiles/CubicSDR.dir/src/ui/UITestCanvas.cpp.o
[100%] Building CXX object CMakeFiles/CubicSDR.dir/src/ui/UITestContext.cpp.o
Linking CXX executable x86/CubicSDR
[100%] Built target CubicSDR
```

Then I executed the binary just to get output for you too.


```
~/Downloads/SDR/CubicSDR-VR2/build$ ls
CMakeCache.txt  cmake_install.cmake    CubicSDR.desktop  Makefile
CMakeFiles      cmake_uninstall.cmake  deb_post.sh       x86
~/Downloads/SDR/CubicSDR-VR2/build$ cd x86
~/Downloads/SDR/CubicSDR-VR2/build/x86$ ls
CubicSDR  CubicSDR.ico  fonts
~/Downloads/SDR/CubicSDR-VR2/build/x86$ ./CubicSDR 
Loading:: configuration file '~/.CubicSDR/config.xml'

Audio Device #0 PulseAudio
	Default Output? Yes
	Default Input? Yes
	Input channels: 2
	Output channels: 2
	Duplex channels: 2
	Native formats:
		16-bit signed integer.
		32-bit signed integer.
		32-bit float normalized between plus/minus 1.0.
	Supported sample rates:
		8000hz
		16000hz
		22050hz
		32000hz
		44100hz
		48000hz
		96000hz

Available vertical sync SwapInterval functions: 
	glxSwapIntervalEXT: No
	DRI2SwapInterval: No
	glxSwapIntervalMESA: No
	glxSwapIntervalSGI: No
No vertical sync swap interval functions available.
Loaded font 'Bitstream Vera Sans Mono' from '~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts/vera_sans_mono12_0.png', parsed 255 characters.
Loaded font 'Bitstream Vera Sans Mono' from '~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts/vera_sans_mono16_0.png', parsed 255 characters.
Loaded font 'Bitstream Vera Sans Mono' from '~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts/vera_sans_mono18_0.png', parsed 255 characters.
Loaded font 'Bitstream Vera Sans Mono' from '~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts/vera_sans_mono24_0.png', parsed 255 characters.
Loaded font 'Bitstream Vera Sans Mono' from '~/Downloads/SDR/CubicSDR-VR2/build/x86/fonts/vera_sans_mono27_0.png', parsed 255 characters.
SDR enumerator starting.
SoapySDR init..
	API Version: v0.6.0
	ABI Version: v0.6
	Install root: /usr/local
	Loading modules... 
	Available factories...null, remote, sdrplay
[WARNING] SoapySSDPEndpoint failed join group udp://[ff02::c]:1900
  setsockopt(IPV6_ADD_MEMBERSHIP) [19: No such device]

Reporting enumeration complete.
SDR enumerator done.
ERROR: thread '9SDRThread' has not terminated in time ! (> 2000 ms)
SDR enumerator starting.
SoapySDR init..
	API Version: v0.6.0
	ABI Version: v0.6
	Install root: /usr/local
	Loading modules... 
	Available factories...null, remote, sdrplay

Reporting enumeration complete.
SDR enumerator done.
Terminating SDR thread..
ERROR: thread '9SDRThread' has not terminated in time ! (> 3000 ms)
Terminating SDR post-processing thread..
Terminating All Demodulators..
Terminating Visual Processor threads..
```

And here is a screenshot.

![screenshot2017-05-28 11 02 30](https://cloud.githubusercontent.com/assets/12898819/26529403/c642d92e-4395-11e7-8320-48301f90ce66.jpg)

We are almost done.  Please see below:

```
~/Downloads/SDR/CubicSDR-VR2_2/build$ make
Scanning dependencies of target CubicSDR
[  1%] Building CXX object CMakeFiles/CubicSDR.dir/external/rtaudio/RtAudio.cpp.o
[  2%] Building CXX object CMakeFiles/CubicSDR.dir/src/CubicSDR.cpp.o
[  3%] Building CXX object CMakeFiles/CubicSDR.dir/src/AppFrame.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/AppFrame.cpp:1330:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 bool AppFrame::actionOnMenuRig(wxCommandEvent& event) {
      ^
~/Downloads/SDR/CubicSDR-VR2_2/src/AppFrame.cpp:2010:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void AppFrame::OnAboutDialogClose(wxCommandEvent& event) {
      ^
~/Downloads/SDR/CubicSDR-VR2_2/src/AppFrame.cpp:2579:6: warning: unused parameter ‘center_freq’ [-Wunused-parameter]
 void AppFrame::setViewState(long long center_freq) {
      ^
~/Downloads/SDR/CubicSDR-VR2_2/src/AppFrame.cpp: In function ‘std::vector<std::basic_string<char> > str_explode(const string&, const string&)’:
~/Downloads/SDR/CubicSDR-VR2_2/src/AppFrame.cpp:2611:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (j == std::string::npos && i < str_len)  j = str_len;
                               ^
~/Downloads/SDR/CubicSDR-VR2_2/src/AppFrame.cpp:2613:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (j == std::string::npos) break;
                               ^
[  5%] Building CXX object CMakeFiles/CubicSDR.dir/src/AppConfig.cpp.o
[  6%] Building CXX object CMakeFiles/CubicSDR.dir/src/FrequencyDialog.cpp.o
[  7%] Building CXX object CMakeFiles/CubicSDR.dir/src/DemodLabelDialog.cpp.o
[  8%] Building CXX object CMakeFiles/CubicSDR.dir/src/IOThread.cpp.o
[ 10%] Building CXX object CMakeFiles/CubicSDR.dir/src/ModemProperties.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/ModemProperties.cpp: In member function ‘void ModemProperties::initDefaultProperties()’:
~/Downloads/SDR/CubicSDR-VR2_2/src/ModemProperties.cpp:76:13: warning: variable ‘defaultDevice’ set but not used [-Wunused-but-set-variable]
         int defaultDevice = 0;
             ^
~/Downloads/SDR/CubicSDR-VR2_2/src/ModemProperties.cpp: At global scope:
~/Downloads/SDR/CubicSDR-VR2_2/src/ModemProperties.cpp:264:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ModemProperties::OnCollapse(wxPropertyGridEvent &event) {
      ^
~/Downloads/SDR/CubicSDR-VR2_2/src/ModemProperties.cpp:268:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ModemProperties::OnExpand(wxPropertyGridEvent &event) {
      ^
[ 11%] Building CXX object CMakeFiles/CubicSDR.dir/src/BookmarkMgr.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/BookmarkMgr.cpp:502:18: warning: unused parameter ‘name_in’ [-Wunused-parameter]
 BookmarkEntryPtr BookmarkMgr::nodeToBookmark(const char *name_in, DataNode *node) {
                  ^
[ 12%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SDRDeviceInfo.cpp.o
[ 13%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SDRPostThread.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/sdr/SDRPostThread.cpp: In member function ‘void SDRPostThread::updateActiveDemodulators()’:
~/Downloads/SDR/CubicSDR-VR2_2/src/sdr/SDRPostThread.cpp:98:38: warning: unused variable ‘demodQueue’ [-Wunused-variable]
         DemodulatorThreadInputQueue *demodQueue = demod->getIQInputDataPipe();
                                      ^
[ 15%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SDREnumerator.cpp.o
[ 16%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorPreThread.cpp.o
[ 17%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorThread.cpp.o
[ 18%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorWorkerThread.cpp.o
[ 20%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorInstance.cpp.o
[ 21%] Building CXX object CMakeFiles/CubicSDR.dir/src/demod/DemodulatorMgr.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/demod/DemodulatorMgr.cpp: In member function ‘DemodulatorInstance* DemodulatorMgr::loadInstance(DataNode*)’:
~/Downloads/SDR/CubicSDR-VR2_2/src/demod/DemodulatorMgr.cpp:529:10: warning: variable ‘found_device’ set but not used [-Wunused-but-set-variable]
     bool found_device = false;
          ^
[ 22%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/Modem.cpp.o
[ 23%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/ModemAnalog.cpp.o
[ 25%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/ModemDigital.cpp.o
[ 26%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemAM.cpp.o
[ 27%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemDSB.cpp.o
[ 28%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemFM.cpp.o
[ 30%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemNBFM.cpp.o
[ 31%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemFMStereo.cpp.o
[ 32%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemIQ.cpp.o
[ 33%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemLSB.cpp.o
[ 35%] Building CXX object CMakeFiles/CubicSDR.dir/src/modules/modem/analog/ModemUSB.cpp.o
[ 36%] Building CXX object CMakeFiles/CubicSDR.dir/src/audio/AudioThread.cpp.o
[ 37%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/Gradient.cpp.o
[ 38%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/Timer.cpp.o
[ 40%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/MouseTracker.cpp.o
[ 41%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/GLExt.cpp.o
[ 42%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/GLFont.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/util/GLFont.cpp:15:21: warning: ‘std::wstring getExePath()’ defined but not used [-Wunused-function]
 static std::wstring getExePath(void)
                     ^
[ 43%] Building CXX object CMakeFiles/CubicSDR.dir/src/util/DataTree.cpp.o
[ 45%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/ScopePanel.cpp.o
[ 46%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/SpectrumPanel.cpp.o
[ 47%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/WaterfallPanel.cpp.o
[ 48%] Building CXX object CMakeFiles/CubicSDR.dir/src/panel/MeterPanel.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/panel/MeterPanel.cpp:129:7: warning: unused parameter ‘panel’ [-Wunused-parameter]
 float MeterPanel::getMeterHitValue(CubicVR::vec2 mousePoint, GLPanel &panel) {
       ^
[ 50%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ColorTheme.cpp.o
[ 51%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/PrimaryGLContext.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/visual/PrimaryGLContext.cpp: In member function ‘void PrimaryGLContext::DrawDemod(DemodulatorInstance*, RGBA4f, long long int, long long int)’:
~/Downloads/SDR/CubicSDR-VR2_2/src/visual/PrimaryGLContext.cpp:359:19: warning: variable ‘demodAlign’ set but not used [-Wunused-but-set-variable]
     GLFont::Align demodAlign = GLFont::GLFONT_ALIGN_CENTER;
                   ^
[ 52%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/InteractiveCanvas.cpp.o
[ 53%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/MeterCanvas.cpp.o
[ 55%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/MeterContext.cpp.o
[ 56%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/TuningCanvas.cpp.o
[ 57%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/TuningContext.cpp.o
[ 58%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ModeSelectorCanvas.cpp.o
[ 60%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ModeSelectorContext.cpp.o
[ 61%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ScopeCanvas.cpp.o
[ 62%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ScopeContext.cpp.o
[ 63%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/SpectrumCanvas.cpp.o
[ 65%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/WaterfallCanvas.cpp.o
[ 66%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/GainCanvas.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/visual/GainCanvas.cpp: In member function ‘void GainCanvas::updateGainUI()’:
~/Downloads/SDR/CubicSDR-VR2_2/src/visual/GainCanvas.cpp:206:16: warning: deleting object of polymorphic class type ‘MeterPanel’ which has non-virtual destructor might cause undefined behaviour [-Wdelete-non-virtual-dtor]
         delete mDel;
                ^
[ 67%] Building CXX object CMakeFiles/CubicSDR.dir/src/visual/ImagePanel.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/visual/ImagePanel.cpp:13:6: warning: unused parameter ‘evt’ [-Wunused-parameter]
 void ImagePanel::paintEvent(wxPaintEvent & evt) {
      ^
[ 68%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/VisualProcessor.cpp.o
[ 70%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/ScopeVisualProcessor.cpp.o
[ 71%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/SpectrumVisualProcessor.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/process/SpectrumVisualProcessor.cpp: In member function ‘virtual void SpectrumVisualProcessor::process()’:
~/Downloads/SDR/CubicSDR-VR2_2/src/process/SpectrumVisualProcessor.cpp:418:17: warning: ‘bwDiff’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 if (bwDiff < 0) {
                 ^
[ 72%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/FFTVisualDataThread.cpp.o
[ 73%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/FFTDataDistributor.cpp.o
[ 75%] Building CXX object CMakeFiles/CubicSDR.dir/src/process/SpectrumVisualDataThread.cpp.o
[ 76%] Building CXX object CMakeFiles/CubicSDR.dir/src/ui/GLPanel.cpp.o
[ 77%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDevices.cpp.o
[ 78%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDevicesForm.cpp.o
[ 80%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDeviceAdd.cpp.o
[ 81%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/SDRDevices/SDRDeviceAddForm.cpp.o
[ 82%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Bookmark/BookmarkPanel.cpp.o
[ 83%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Bookmark/BookmarkView.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Bookmark/BookmarkView.cpp:676:11: warning: unused parameter ‘parent’ [-Wunused-parameter]
 wxButton *BookmarkView::makeButton(wxWindow *parent, std::string labelVal, wxObjectEventFunction handler) {
           ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Bookmark/BookmarkView.cpp:1475:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void BookmarkView::onSearchText( wxCommandEvent& event ) {
      ^
[ 85%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/ActionDialogBase.cpp.o
[ 86%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/ActionDialog.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/ActionDialog.cpp:39:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ActionDialog::onClickCancel( wxCommandEvent& event ) {
      ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/ActionDialog.cpp:48:6: warning: unused parameter ‘event’ [-Wunused-parameter]
 void ActionDialog::onClickOK( wxCommandEvent& event ) {
      ^
[ 87%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/AboutDialogBase.cpp.o
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp: In constructor ‘AboutDialogBase::AboutDialogBase(wxWindow*, wxWindowID, const wxString&, const wxPoint&, const wxSize&, long int)’:
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:25:67: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_appName->SetFont( wxFont( 20, 70, 90, 90, false, wxEmptyString ) );
                                                                   ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:48:68: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dbHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                    ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:54:97: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dbGHHeader->SetFont( wxFont( wxNORMAL_FONT->GetPointSize(), 70, 90, 90, false, wxEmptyString ) );
                                                                                                 ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:60:96: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dbTwitter->SetFont( wxFont( wxNORMAL_FONT->GetPointSize(), 70, 90, 90, false, wxEmptyString ) );
                                                                                                ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:101:79: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_cContributorsHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                               ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:107:94: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_cGitHub->SetFont( wxFont( wxNORMAL_FONT->GetPointSize(), 70, 90, 90, false, wxEmptyString ) );
                                                                                              ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:193:67: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_dHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                   ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:386:68: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stHeader->SetFont( wxFont( 15, 70, 90, 90, false, wxEmptyString ) );
                                                                    ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:395:82: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stSoapyDevAssistHeader->SetFont( wxFont( 10, 70, 90, 92, false, wxEmptyString ) );
                                                                                  ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:408:77: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stLiquidDSPHeader->SetFont( wxFont( 10, 70, 90, 92, false, wxEmptyString ) );
                                                                             ^
~/Downloads/SDR/CubicSDR-VR2_2/src/forms/Dialog/AboutDialogBase.cpp:421:83: warning: ‘wxFont::wxFont(int, int, int, int, bool, const wxString&, wxFontEncoding)’ is deprecated (declared at ~/Downloads/SDR/Dev/wxWidgets-staticlib/include/wx-3.1/wx/gtk/font.h:89): use wxFONT{FAMILY,STYLE,WEIGHT}_XXX constants [-Wdeprecated-declarations]
  m_stIdeasDirectionsHeader->SetFont( wxFont( 10, 70, 90, 92, false, wxEmptyString ) );
                                                                                   ^
[ 88%] Building CXX object CMakeFiles/CubicSDR.dir/src/forms/Dialog/AboutDialog.cpp.o
[ 90%] Building CXX object CMakeFiles/CubicSDR.dir/external/lodepng/lodepng.cpp.o
[ 91%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinyxml.cpp.o
[ 92%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinystr.cpp.o
[ 93%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinyxmlparser.cpp.o
[ 95%] Building CXX object CMakeFiles/CubicSDR.dir/external/tinyxml/tinyxmlerror.cpp.o
[ 96%] Building CXX object CMakeFiles/CubicSDR.dir/external/cubicvr2/math/cubic_math.cpp.o
[ 97%] Building CXX object CMakeFiles/CubicSDR.dir/src/sdr/SoapySDRThread.cpp.o
[ 98%] Building CXX object CMakeFiles/CubicSDR.dir/src/ui/UITestCanvas.cpp.o
[100%] Building CXX object CMakeFiles/CubicSDR.dir/src/ui/UITestContext.cpp.o
Linking CXX executable x86/CubicSDR
[100%] Built target CubicSDR
```

--- a/external/cubicvr2/math/mat3.h
+++ b/external/cubicvr2/math/mat3.h
@@ -12,7 +12,7 @@
 #include <iostream>
 #include "vec3.h"
 #include <cstring>
-
+#include <stddef.h>
 namespace CubicVR {
 
     #define mat3SG(c,x,y) \
@@ -20,14 +20,20 @@ namespace CubicVR {
         c & COMBINE(set,x)(mat3 value) { y = value; return *this; }
 
     struct mat3 {
+       
         __float a,b,c,d,e,f,g,h,i;
-
-        //        __float  operator [] (unsigned i) const { return ((__float *)this)[i]; }
-#ifndef _WIN32
-        __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
-#endif
-        operator __float*() const { return (__float *)this; }
+      
+        //access as-array:
+        inline __float& operator [] (size_t i) {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
         
+        inline const __float& operator [] (size_t i) const { 
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
+    
         mat3(__float ai,__float bi,__float ci,__float di,__float ei,__float fi,__float gi,__float hi,__float ii) {
             a = ai; b = bi; c = ci; d = di; e = ei; f = fi; g = gi; h = hi; i = ii;
         };

--- a/external/cubicvr2/math/mat4.h
+++ b/external/cubicvr2/math/mat4.h
@@ -15,21 +15,44 @@
 #include "vec4.h"
 #include "mat3.h"
 #include <cmath>
+#include <stddef.h>
 
 namespace CubicVR {
     using namespace std;
     #define mat4SG(c,x,y) \
         mat4 COMBINE(get,x)() { return y; } \
         c & COMBINE(set,x)(mat4 value) { y = value; return *this; }
+
     struct mat4 {
-        __float a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p;
+      
+         //16 elements
+         __float a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p;
+      
+        //access as-array:
+        inline __float& operator [] (size_t i) {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
 
-        //        __float  operator [] (unsigned i) const { return ((__float *)this)[i]; }
-#ifndef _WIN32
-        __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
-#endif
+        inline const __float& operator [] (size_t i) const {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
 
-        operator __float*() const { return (__float *)this; }
+        //compare to ZERO-filled matrix
+        inline operator bool() const {
+            
+            return (a != 0.0f || b != 0.0f || c != 0.0f || d != 0.0f ||
+                e != 0.0f || f != 0.0f || g != 0.0f || h != 0.0f ||
+                i != 0.0f || j != 0.0f || k != 0.0f || l != 0.0f ||
+                m != 0.0f || n != 0.0f || o != 0.0f || p != 0.0f);
+        }
+
+        //To be accessed by GL API directly, accessed by pointer.
+        //operator* overloading is way too dangerous, especially in ptr != NULL
+        //tests.
+       inline  float* to_ptr() const { return (__float *)this; }
+
         mat4(__float ai,__float bi,__float ci,__float di,__float ei,__float fi,__float gi,__float hi,__float ii,__float ji,__float ki,__float li,__float mi,__float ni,__float oi,__float pi) {
             a = ai; b = bi; c = ci; d = di; e = ei; f = fi; g = gi; h = hi; i = ii; j = ji; k = ki; l = li; m = mi; n = ni; o = oi; p = pi;
         }
@@ -263,15 +286,15 @@ namespace CubicVR {
         static mat4 transform(vec3 position, vec3 rotation, vec3 scale) {
             mat4 m = mat4::identity();
             
-            if (position!=NULL) {
+            if (!position) {
                 m *= mat4::translate(position[0],position[1],position[2]);
             }
-            if (rotation!=NULL) {
+            if (!rotation) {
                 if (!(rotation[0] == 0 && rotation[1] == 0 && rotation[2] == 0)) {
                     m *= mat4::rotate(rotation[0],rotation[1],rotation[2]);
                 }
             }
-            if (scale!=NULL) {
+            if (!scale) {
                 if (!(scale[0] == 1 && scale[1] == 1 && scale[2] == 1)) {
                     m *= mat4::scale(scale[0],scale[1],scale[2]);
                 }

--- a/external/cubicvr2/math/mat4.h
+++ b/external/cubicvr2/math/mat4.h
@@ -39,15 +39,6 @@ namespace CubicVR {
             return (as_array[i]);
         }
 
-        //compare to ZERO-filled matrix
-        inline operator bool() const {
-            
-            return (a != 0.0f || b != 0.0f || c != 0.0f || d != 0.0f ||
-                e != 0.0f || f != 0.0f || g != 0.0f || h != 0.0f ||
-                i != 0.0f || j != 0.0f || k != 0.0f || l != 0.0f ||
-                m != 0.0f || n != 0.0f || o != 0.0f || p != 0.0f);
-        }
-
         //To be accessed by GL API directly, accessed by pointer.
         //operator* overloading is way too dangerous, especially in ptr != NULL
         //tests.
@@ -286,22 +277,19 @@ namespace CubicVR {
         static mat4 transform(vec3 position, vec3 rotation, vec3 scale) {
             mat4 m = mat4::identity();
             
-            if (!position) {
-                m *= mat4::translate(position[0],position[1],position[2]);
+            m *= mat4::translate(position[0],position[1],position[2]);
+          
+            if (!(rotation[0] == 0 && rotation[1] == 0 && rotation[2] == 0)) {
+                m *= mat4::rotate(rotation[0], rotation[1], rotation[2]);    
             }
-            if (!rotation) {
-                if (!(rotation[0] == 0 && rotation[1] == 0 && rotation[2] == 0)) {
-                    m *= mat4::rotate(rotation[0],rotation[1],rotation[2]);
-                }
-            }
-            if (!scale) {
-                if (!(scale[0] == 1 && scale[1] == 1 && scale[2] == 1)) {
-                    m *= mat4::scale(scale[0],scale[1],scale[2]);
-                }
+             
+            if (!(scale[0] == 1 && scale[1] == 1 && scale[2] == 1)) {
+                m *= mat4::scale(scale[0],scale[1],scale[2]);
             }
             
             return m;
         };
+
         static vec4 vec4_multiply(vec4 m1, mat4 m2) {
             vec4 mOut;
 

--- a/external/cubicvr2/math/transform.h
+++ b/external/cubicvr2/math/transform.h
@@ -73,7 +73,7 @@ namespace CubicVR {
         
         void pushMatrix(mat4 m) {
             c_stack++;
-            m_stack[c_stack] = (m ? m : getIdentity());
+            m_stack[c_stack] = m;
         }
         
         void popMatrix() {
@@ -91,11 +91,7 @@ namespace CubicVR {
 
             result = mat4::identity();
             
-            if (!init_mat) {
-                m_stack[0] = init_mat;
-            } else {
-                setIdentity();
-            }
+            m_stack[0] = init_mat; 
         }
         
         void translate(__float x, __float y, __float z) {

--- a/external/cubicvr2/math/transform.h
+++ b/external/cubicvr2/math/transform.h
@@ -88,10 +88,10 @@ namespace CubicVR {
             m_cache.clear();
             c_stack = 0;
             valid = 0;
-            delete result;
+
             result = mat4::identity();
             
-            if (init_mat != NULL) {
+            if (!init_mat) {
                 m_stack[0] = init_mat;
             } else {
                 setIdentity();

--- a/external/cubicvr2/math/vec2.h
+++ b/external/cubicvr2/math/vec2.h
@@ -12,6 +12,7 @@
 #include <iostream>
 #include <cmath>
 #include "cubic_types.h"
+#include <stddef.h>
 
 namespace CubicVR {
     #define vec2SG(c,x,y) \
@@ -20,20 +21,27 @@ namespace CubicVR {
 
     struct vec2 {
         __float x, y;
+
     public:
         __float& u() { return x; }
         __float& v() { return y; }
         
-//        __float  operator [] (unsigned i) const { return ((__float *)this)[i]; }
-#ifndef _WIN32
-		__float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
-#endif        
+        //access as-array:
+        inline __float& operator [] (size_t i) {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
+
+        inline const __float& operator [] (size_t i) const {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
+
         vec2 (__float xi,__float yi) { x = xi; y = yi;  }
         vec2 () { x = y = 0.0f; }
-        
-        operator __float*() const { return (__float *)this; }
-        
+         
         vec2 operator*(__float v) { return vec2( x*v, y*v ); }
+
         //    vec2 operator*(vec2 v) { return vec2::cross(*this,v); }
         vec2 operator+(vec2 v) { return vec2::add(*this,v); }
         vec2 operator-(vec2 v) { return vec2::subtract(*this,v); }

--- a/external/cubicvr2/math/vec3.h
+++ b/external/cubicvr2/math/vec3.h
@@ -39,12 +39,6 @@ namespace CubicVR {
             return (as_array[i]);
         }
 
-        //compare to ZERO-filled vector
-        inline operator bool() const {
-
-            return (x != 0.0f || y != 0.0f || z != 0.0f);
-        }
-
         vec3 (__float xi,__float yi,__float zi) { x = xi; y = yi; z = zi; }
         vec3 () { x = y = z = 0.0f; }
         

--- a/external/cubicvr2/math/vec3.h
+++ b/external/cubicvr2/math/vec3.h
@@ -12,6 +12,7 @@
 #include <iostream>
 #include "cubic_types.h"
 #include <cmath>
+#include <stddef.h>
 
 namespace CubicVR {
     
@@ -23,18 +24,29 @@ namespace CubicVR {
     struct vec3 {
         __float x,y,z;
 
-        operator __float*() const { return (__float *)this; }
-
         __float& r() { return x; }
         __float& g() { return y; }
         __float& b() { return z; }
         
-#ifndef _WIN32
-        __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
-#endif
+        //access as-array:
+        inline __float& operator [] (size_t i) {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
+
+        inline const __float& operator [] (size_t i) const {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
+
+        //compare to ZERO-filled vector
+        inline operator bool() const {
+
+            return (x != 0.0f || y != 0.0f || z != 0.0f);
+        }
+
         vec3 (__float xi,__float yi,__float zi) { x = xi; y = yi; z = zi; }
         vec3 () { x = y = z = 0.0f; }
-        
         
         vec3 operator*(__float v) { return vec3(x*v, y*v, z*v); }
         vec3 operator*(vec3 v) { return vec3::cross(*this,v); }

--- a/external/cubicvr2/math/vec4.h
+++ b/external/cubicvr2/math/vec4.h
@@ -12,6 +12,7 @@
 #include <iostream>
 #include "cubic_types.h"
 #include <cmath>
+#include <stddef.h>
 
 namespace CubicVR {
 
@@ -20,22 +21,28 @@ namespace CubicVR {
     c & COMBINE(set,x)(vec3 value) { y = value; return *this; }
 
     struct vec4 {
-        __float x,y,z,w;
+
+        __float x, y, z, w;
+
     public:
         __float& r() { return x; }
         __float& g() { return y; }
         __float& b() { return z; }
         __float& a() { return w; }
         
-//        __float  operator [] (unsigned i) const { return ((__float *)this)[i]; }
-#ifndef _WIN32
-        __float& operator [] (unsigned i)       { return ((__float *)this)[i]; }
-#endif
+        //access as-array:
+        inline __float& operator [] (size_t i) {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
+
+        inline const __float& operator [] (size_t i) const {
+            __float* as_array = (__float*)this;
+            return (as_array[i]);
+        }
 
         vec4 (__float xi,__float yi,__float zi,__float wi) { x = xi; y = yi; z = zi; w = wi; }
         vec4 () { x = y = z = w =  0.0f; }
-        
-        operator __float*() const { return (__float *)this; }
         
         vec4 operator*(__float v) { return vec4(x*v, y*v, z*v, w*v); }
 //        vec4 operator*(vec4 v) { return vec4::cross(*this,v); }

--- a/src/panel/ScopePanel.cpp
+++ b/src/panel/ScopePanel.cpp
@@ -35,7 +35,7 @@ void ScopePanel::drawPanelContents() {
         bgPanel.draw();
         glLineWidth(1.0);
         glEnable(GL_LINE_SMOOTH);
-        glLoadMatrixf(transform);
+        glLoadMatrixf(transform.to_ptr());
         glColor3f(ThemeMgr::mgr.currentTheme->scopeLine.r * 0.35, ThemeMgr::mgr.currentTheme->scopeLine.g * 0.35,
                   ThemeMgr::mgr.currentTheme->scopeLine.b * 0.35);
         glBegin (GL_LINES);
@@ -52,7 +52,7 @@ void ScopePanel::drawPanelContents() {
         bgPanelStereo[1].draw();
 
         glLineWidth(1.0);
-        glLoadMatrixf(transform);
+        glLoadMatrixf(transform.to_ptr());
         glColor3f(ThemeMgr::mgr.currentTheme->scopeLine.r, ThemeMgr::mgr.currentTheme->scopeLine.g, ThemeMgr::mgr.currentTheme->scopeLine.b);
         glEnable(GL_LINE_SMOOTH);
         glBegin (GL_LINES);
@@ -76,7 +76,7 @@ void ScopePanel::drawPanelContents() {
         glLineWidth(1.0);
         glEnable(GL_POINT_SMOOTH);
         glPointSize(1.0);
-        glLoadMatrixf(transform);
+        glLoadMatrixf(transform.to_ptr());
         glColor3f(ThemeMgr::mgr.currentTheme->scopeLine.r * 0.15, ThemeMgr::mgr.currentTheme->scopeLine.g * 0.15,
                   ThemeMgr::mgr.currentTheme->scopeLine.b * 0.15);
     }
@@ -95,16 +95,16 @@ void ScopePanel::drawPanelContents() {
         glVertexPointer(2, GL_FLOAT, 0, &points[0]);
         glLineWidth(1.5);
         if (scopeMode == SCOPE_MODE_Y) {
-            glLoadMatrixf(bgPanel.transform);
+            glLoadMatrixf(bgPanel.transform.to_ptr());
             glDrawArrays(GL_LINE_STRIP, 0, points.size() / 2);
         } else if (scopeMode == SCOPE_MODE_2Y)  {
-            glLoadMatrixf(bgPanelStereo[0].transform);
+            glLoadMatrixf(bgPanelStereo[0].transform.to_ptr());
             glDrawArrays(GL_LINE_STRIP, 0, points.size() / 4);
 
-            glLoadMatrixf(bgPanelStereo[1].transform);
+            glLoadMatrixf(bgPanelStereo[1].transform.to_ptr());
             glDrawArrays(GL_LINE_STRIP, points.size() / 4, points.size() / 4);
         } else if (scopeMode == SCOPE_MODE_XY) {
-            glLoadMatrixf(bgPanel.transform);
+            glLoadMatrixf(bgPanel.transform.to_ptr());
             glDrawArrays(GL_POINTS, 0, points.size() / 2);
         }
         glLineWidth(1.0);

--- a/src/panel/SpectrumPanel.cpp
+++ b/src/panel/SpectrumPanel.cpp
@@ -113,7 +113,7 @@ void SpectrumPanel::drawPanelContents() {
     glEnable(GL_LINE_SMOOTH);
     glHint( GL_LINE_SMOOTH_HINT, GL_NICEST );
 
-    glLoadMatrixf(transform * (CubicVR::mat4::translate(-1.0f, -0.75f, 0.0f) * CubicVR::mat4::scale(2.0f, 1.5f, 1.0f)));
+    glLoadMatrixf((transform * (CubicVR::mat4::translate(-1.0f, -0.75f, 0.0f) * CubicVR::mat4::scale(2.0f, 1.5f, 1.0f))).to_ptr());
 
     if (points.size()) {
         glBlendFunc(GL_SRC_ALPHA, GL_ONE);
@@ -166,7 +166,7 @@ void SpectrumPanel::drawPanelContents() {
     
     float viewHeight = (float) vp[3];
     float viewWidth = (float) vp[2];
-    glLoadMatrixf(transform);
+    glLoadMatrixf(transform.to_ptr());
 
     
     long long leftFreq = (double) freq - ((double) bandwidth / 2.0);

--- a/src/panel/WaterfallPanel.cpp
+++ b/src/panel/WaterfallPanel.cpp
@@ -165,7 +165,7 @@ void WaterfallPanel::drawPanelContents() {
 
     int half_fft_size = fft_size / 2;
     
-    glLoadMatrixf(transform);
+    glLoadMatrixf(transform.to_ptr());
     
     glEnable (GL_TEXTURE_2D);
     glTexEnvf(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_DECAL);

--- a/src/ui/GLPanel.cpp
+++ b/src/ui/GLPanel.cpp
@@ -308,7 +308,7 @@ void GLPanel::calcTransform(mat4 transform_in) {
 void GLPanel::draw() {
     float min = -1.0, max = 1.0;
 
-    glLoadMatrixf(transform);
+    glLoadMatrixf(transform.to_ptr());
     
     if (fillType != GLPANEL_FILL_NONE && visible) {
         glEnable(GL_BLEND);
@@ -378,7 +378,7 @@ void GLPanel::draw() {
         }
         //        if (coord == GLPANEL_Y_UP) {
         //        }
-        glLoadMatrixf(transform * mCoord);
+        glLoadMatrixf((transform * mCoord).to_ptr());
         drawPanelContents();
     }
 }

--- a/src/visual/ScopeCanvas.cpp
+++ b/src/visual/ScopeCanvas.cpp
@@ -148,7 +148,7 @@ void ScopeCanvas::OnPaint(wxPaintEvent& WXUNUSED(event)) {
     
     glMatrixMode(GL_PROJECTION);
     glLoadIdentity();
-    glLoadMatrixf(CubicVR::mat4::perspective(45.0, 1.0, 1.0, 1000.0));
+    glLoadMatrixf(CubicVR::mat4::perspective(45.0, 1.0, 1.0, 1000.0).to_ptr());
     glMatrixMode(GL_MODELVIEW);
     glLoadIdentity();
     
@@ -210,7 +210,7 @@ void ScopeCanvas::OnPaint(wxPaintEvent& WXUNUSED(event)) {
         spectrumPanel.drawChildren();
     }
     
-    glLoadMatrixf(scopePanel.transform);
+    glLoadMatrixf(scopePanel.transform.to_ptr());
     if (!deviceName.empty()) {
         glContext->DrawDeviceName(deviceName);
     }

